doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,145 @@
+
+PMON - Platform Monitor container in SONiC.
+
+PSU - Power Supply Unit
+
+Voltage Sensor - Sensor device which can report a voltage measurement in the system.
+
+Current Sensor - Sensor device which can report current measurement in the system.
+
+Altitude Sensor - Sensor device which can report the altitude of the system.
+
+
+### Overview 
+
+Modern hardware systems have many different types of sensors and control devices. Voltage sensor devices can measure and in some cases control the voltages on the boards. Current sensor devices can measure current. It is also possible to have other types of sensors such as Altitude Sensors etc. These devices can report measurements from different parts of the system which are useful for monitoring system health. For example, voltage controller devices distribute power across different parts of the system such as  motherboard, daughterboards etc. and can report voltage measurements from there. Often these devices can report under-voltage/over-voltage faults which should be monitored to alert the operator about any power related failures in the system. This document provides an overview for monitoring voltage and current sensors in SONiC. The solution proposed in this document can be enhanced for other types of sensors as well.
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+
+### Requirements
+
+This HLD covers
+
+* Discovery of voltage and current sensor devices in the system
+* Monitoring the sensor devices periodically and update that data in Redis DB
+* Raise/Clear alarms if the sensor devices indicate readings which are unexpected and clear them if they return to a good state.
+* A framework for adding new sensor types in future
+
+This HLD does not cover
+
+* An automated recovery action that system might take as a result of a fault reported by the voltage sensor device. A network management system may process the alarms and take recvoery action as it sees fit.
+
+
+### High Level Design 
+
+The proposal for monitoring sensor devices is to create a new SensorMon daemon. SensorMon will use API provided by platform to discover the sensor devices. It will periodically poll the devices to refresh the sensor information and update the readings in StateDB.
+
+If the sensor device readings cross the minor/major/critical thresholds, syslogs will be generated to indicate to the operator about the alarm condition. If the sensor reports normal data in a subsequent poll, another syslog will be generated to indicate that the fault condition is cleared.
+
+CLIs are provided to display the sensor devices, their measurements, threshold values and if they are reporting an alarm.
+
+Platform APIs will provide 
+
+* List of sensors devices of a specific type in the system
+* Way to read the sensor information
```

Comment:
@bmridul can we make use of exisiting `sensors` command ?

Reply:
As mentioned in the other comment, you can use sensors for the hwmon supported sensors if platform vendor prefers that. Sensors.conf provides flexibility if you want the sensor to be monitored by sensord or not.